staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -1331,10 +1331,16 @@ func (r Result) Into(obj runtime.Object) error {
 			r.statusCode, r.contentType)
 	}
 
-	out, _, err := r.decoder.Decode(r.body, nil, obj)
-	if err != nil || out == obj {
+	out, gvk, err := r.decoder.Decode(r.body, nil, obj)
+	if err != nil {
 		return err
 	}
+	if out == obj {
+		if gvk != nil {
+			obj.GetObjectKind().SetGroupVersionKind(*gvk)
+		}
+		return nil
+	}
```

Comment:
We should fix this by stop [removing GVK in the decoder](https://github.com/kubernetes/kubernetes/blob/dd5c3a109e2f156774576b7e7a38e104fa5c7202/staging/src/k8s.io/apimachinery/pkg/runtime/helper.go#L254-L256), rather than removing the GVK in the decoder and then putting it back here. This seems to be what @liggitt suggested in https://github.com/kubernetes/kubernetes/issues/80609 as well.